tests/Unit/Dashboard/Infrastructure/Score_Groups/Get_View_Link_Score_Group_Link_Collector_Test.php

```diff
@@ -0,0 +1,158 @@
+<?php
+// phpcs:disable Yoast.NamingConventions.NamespaceName.TooLong -- Needed in the folder structure.
+namespace Yoast\WP\SEO\Tests\Unit\Dashboard\Infrastructure\Score_Groups;
+
+use Brain\Monkey\Functions;
+use Generator;
+use Mockery;
+use WP_Taxonomy;
+use WP_Term;
+use Yoast\WP\SEO\Dashboard\Domain\Content_Types\Content_Type;
+use Yoast\WP\SEO\Dashboard\Domain\Score_Groups\Score_Groups_Interface;
+use Yoast\WP\SEO\Dashboard\Domain\Score_Groups\SEO_Score_Groups\Ok_SEO_Score_Group;
+use Yoast\WP\SEO\Dashboard\Domain\Taxonomies\Taxonomy;
+use Yoast\WP\SEO\Dashboard\Infrastructure\Score_Groups\Score_Group_Link_Collector;
+use Yoast\WP\SEO\Tests\Unit\TestCase;
+
+/**
+ * Test class for the get_view_link method.
+ *
+ * @group  Score_Group_Link_Collector
+ *
+ * @covers Yoast\WP\SEO\Dashboard\Infrastructure\Score_Groups\Score_Group_Link_Collector::get_view_link
+ *
+ * @phpcs:disable Yoast.NamingConventions.ObjectNameDepth.MaxExceeded
+ */
+final class Get_View_Link_Score_Group_Link_Collector_Test extends TestCase {
+
+	/**
+	 * Holds the instance.
+	 *
+	 * @var Score_Group_Link_Collector
+	 */
+	private $instance;
+
+	/**
+	 * Sets up the test fixtures.
+	 *
+	 * @return void
+	 */
+	public function set_up() {
+		$this->instance = new Score_Group_Link_Collector();
+	}
+
+	/**
+	 * Tests if the correct view link is generated.
+	 *
+	 * @dataProvider generate_get_view_link_provider
+	 *
+	 * @param string|null            $expected           The expected value.
+	 * @param Score_Groups_Interface $score_group        The score groups the link is generated for.
+	 * @param Content_Type           $content_type       The content type.
+	 * @param Taxonomy|null          $taxonomy           The taxonomy of the term we might be filtering.
+	 * @param string|bool            $taxonomy_query_var The taxonomy query var portion of the taxonomy.
+	 * @param int|null               $term_id            The ID of the term we might be filtering.
+	 *
+	 * @return void
+	 */
+	public function test_get_view_link(
+		?string $expected,
+		Score_Groups_Interface $score_group,
+		Content_Type $content_type,
+		?Taxonomy $taxonomy,
+		$taxonomy_query_var,
+		?int $term_id
+	) {
+		$post_page = 'edit.php';
+		Functions\expect( 'admin_url' )
+			->with( 'edit.php' )
+			->andReturn( $post_page );
+
+		if ( $taxonomy === null || $term_id === null ) {
```

Comment:
Usually it's not recommended to add these conditions in the tests but rather have separate tests for each case, then when one is failing we will get a more specific indication when things went wrong.